contracts/PinningManager.sol

```diff
@@ -191,16 +184,11 @@ contract PinningManager {
             uint256 toTransfer = (request.numberOfPeriodsDeposited - request.numberOfPeriodsWithdrawn).mul(request.chosenPrice);
```

Comment:
What is the reason for this if-branch? Why is there `EarningsWithdrawn` in `newRequest` call?

Reply:
Good question. The reason it is there is because a requestor might have had a request with this provider before. Let's say, this request ended a month ago, and he wants to renew it. 

This if-statement makes sure that the new request starts with a clean state and that the provider is given all the money he should be given, based on the old contract. If the provider has already redeemed all his money, the if-statement won't execute (see lines 350-359, in function withdrawEarnings). But as the storageProvider is not guaranteed to do this, we need to force the accounting if it's not done yet here.

Reply:
I see that is actually very good thinking!

Could you please put this explanation as comment to the code? Because I could not think of this reason just from the code itself...